src/chapters/more-on-functions/recursion-walkthrough2.rst

```diff
@@ -0,0 +1,133 @@
+
+.. figure:: figures/second-case-recursion.png
+   :alt: Visual representation for the second-easiest case.
+
+In this case, ``combineEntries`` executes the ``else`` statement. We have no
+code for this yet, but we can still consider the logic:
+
+a. ``combineEntries`` returns ``'L'`` and calls itself again using what is left
+   inside the array (``['C']``).
+b. When passed ``['C']``, which is the base case, ``combineEntries`` returns
+   ``'C'``.
+c. The strings ``'L'`` and ``'C'`` get combined and returned as the final
+   result.
+
+Next, consider an array with three elements ``['L', 'C', '1']``:
+
+.. figure:: figures/third-case-recursion.png
+   :alt: Visual representation for the third-easiest case.
+
+As before, ``combineEntries`` executes the ``else`` statement, and we can
+follow the logic:
+
+a. ``combineEntries`` returns ``'L'`` and calls itself again using what is left
+   inside the array (``['C', '1']``).
+b. When passed ``['C', '1']``, ``combineEntries`` returns ``'C'`` and calls
+   itself again using what is left inside the array (``['1']``).
+c. When passed ``['1']``, which is the base case, ``combineEntries`` returns
+   ``'1'``.
+d. The strings ``'C'`` and ``'1'`` get combined and returned.
+e. The strings ``'L'`` and ``'C1'`` get combined and returned as the final
+   result.
+
+As we make the array longer, ``combineEntries`` calls itself more times. Each
+call evaluates a smaller and smaller section of the array until reaching the
+base case. This sets up a series of return events - each one selecting a
+single entry from the array. Rather than building ``'LC101'`` from left to
+right, recursion constructs the string starting with the base case and
+adding new characters to the front:
+
+| ``'1'``
+| ``'01'``
+| ``'101'``
+| ``'C101'``
+| ``'LC101'``
+
+A Function Calls Itself
+------------------------
+
+So how do we code the ``else`` statement in ``combineEntries``? Recall what
+needs to happen each time the statement runs:
+
+a. Select the first element in the array,
+b. Call ``combineEntries`` again with a smaller array.
+
+Bracket notation takes care of part a: ``arrayName[0]``.
+
+| For part b, remember that the :ref:`slice method <slice-examples>` returns
+   selected entries from an array. To return everything BUT the first entry in
+| ``arr = ['L', 'C', '1', '0', '1']``, use ``arr.slice(1)``.
+
+Let's add the bracket notation and the ``slice`` method to our function:
+
+.. sourcecode:: js
+
+   function combineEntries(arrayName){
+      if (arrayName.length === 1){
+         return arrayName[0];
+      } else {
+         return arrayName[0]+combineEntries(arrayName.slice(1));
+      }
+   }
+
+Each time the ``else`` statement runs, it extracts the first element in the
+array with ``arrayName[0]``, then it calls itself with the remaining array
+elements (``arrayName.slice(1)``). For ``combineElements(['L', 'C', '1', '0', '1']);``:
+
+| a. First call: Combine ``'L'`` with ``combineEntries(['C', '1', '0', '1'])``.
```

Comment:
Let's chat about this formatting and how to achieve what you are going for. We should avoid hacking the markup language to achieve effects that isn't built for. 

This is a sequential list, which calls for `#.` as the item markers. Manually inserting line breaks means that the generated HTML won't render as a `<ol>` or even a `<ul>`, which can create problems with styling and document reading in various circumstances. 

I know RST doesn't have lettered lists, so my question would be what is the impetus for using lettered lists, and how important is the reasoning in this case?

Reply:
I was going for an "up the chain" visual. The calls to ``combineElements`` move from a --> b --> c --> d --> e, and I wanted to reinforce how LC101 is built from e --> d --> c --> b --> a.

As for lettered lists, I've been using them as sub-bullets for numbered exercises, or as stand-alone lists when I think letters provide more clarity than numbers.  This is simply a style choice.

Reply:
Regarding lettered lists: I'm not opposed to their use if they result in semantic HTML. I don't think Sphinx does this out of the box, but there may be a plugin that could enable the behavior, since CSS supports letters for `ol` elements.

It would also be good to discuss conventions, if we're able to get lettered lists to work properly. Mixing numbered and lettered lists for the same use cases (e.g. exercise sub tasks) would be sloppy.

Reply:
I mix numbers and letters for clarity in exercises.

Ex:
1. Big problem statement.
  a. First sub-task,
  b. Second sub task.

2. New problem.
  a. Etc.
  b. Etc.

I think this works better than:
1. Big problem statement.
  1. First sub-task,
  2. Second sub task.

2. New problem.
  1. Etc.
  2. Etc.

Reply:
Makes sense. Interestingly, the RST docs on the Sphinx site don't cover lettered lists (to my knowledge). As long as they result in `ol` elements in the HTML, I'm happy. 

Maybe we can make your practice a convention then: Ordered sublists of ordered lists are lettered.